simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/securityctrlr/BasicAuthPasswordVariableAccessor.java

```diff
@@ -98,14 +98,18 @@ private SetVariableResult validateActualValue(AttributePath attributePath, CiStr
                 .withVariable(attributePath.getVariable())
                 .withAttributeType(SetVariableResult.AttributeType.fromValue(attributePath.getAttributeType().getName()));
 
-        if (invalidLength(attributeValue) || isNotHex(attributeValue.toString())) {
+        if (invalidLength(attributeValue) || isNotHex(attributeValue.toString()) || isEven(attributeValue)) {
             return setVariableResult.withAttributeStatus(SetVariableResult.AttributeStatus.INVALID_VALUE);
         }
 
         return setVariableResult.withAttributeStatus(SetVariableResult.AttributeStatus.ACCEPTED);
 
     }
 
+    private boolean isEven(CiString.CiString1000 attributeValue) {
+        return (attributeValue.toString().length() & 0x1) == 1;
```

Comment:
👍

Reply:
lol. it's isOdd actually. 
For even numbers length() & 0x1 == 0

Reply:
Even: 2, 4, 50, ..
Odd 1,3, 5, 7,...

Reply:
yes,  s/isEven/isOdd